src/libraries/System.Net.Http/src/System/Net/Http/Headers/CacheControlHeaderValue.cs

```diff
@@ -367,14 +367,14 @@ public static CacheControlHeaderValue Parse(string? input)
             return (CacheControlHeaderValue)CacheControlHeaderParser.Parser.ParseValue(input, null, ref index);
         }
 
-        public static bool TryParse(string? input, out CacheControlHeaderValue? parsedValue)
+        public static bool TryParse(string? input, out CacheControlHeaderValue parsedValue)
         {
             int index = 0;
-            parsedValue = null;
+            parsedValue = new CacheControlHeaderValue();
```

Comment:
```suggestion
            parsedValue = null;
```

This is allocating the value even though it won't be needed in the common case.
The behavior should be such that we set `parsedValue` to something not null if and only if we return true.

The line below can be changed to
```c#
parsedValue = (CacheControlHeaderValue)output ?? new CacheControlHeaderValue();
```

Reply:
The `Parse` method should likewise be updated to never return null.

We already have existing tests for valid/invalid values, but we're missing tests for empty values.
That is, we need a test like
```c#
[Theory]
[InlineData(null)]
[InlineData("")]
[InlineData(" ")]
[InlineData(",")]
[InlineData(", ,")]
public void CacheControlHeaderValue_EmptyValue_Parsed(string value)
{
    Assert.NotNull(CacheControlHeaderValue.Parse(value));

    Assert.True(CacheControlHeaderValue.TryParse(value, out CacheControlHeaderValue headerValue));
    Assert.NotNull(headerValue);
}
```